src/components/ObsEdit/ObsEdit.js

```diff
@@ -113,6 +118,34 @@ const ObsEdit = ( ): Node => {
 
   const currentObs = observations[currentObsIndex];
 
+  useEffect( () => {
+    if ( currentObs ) {
+      const fetchLocation = async () => {
+        const location = await fetchUserLocation( );
+
+        updateObservationKeys( {
+          place_guess: location?.place_guess,
+          latitude: location?.latitude,
+          longitude: location?.longitude,
+          positional_accuracy: location?.positional_accuracy
+        } );
+        if ( location && mounted ) {
+          setPositionalAccuracy( location.positional_accuracy );
+        }
+      };
+
+      if ( !fetchedLocation && !currentObs._created_at ) {
```

Comment:
The `!fetchedLocation` means this only fetches the location when `fetchedLocation` is blank. That means if you fetch it and it sets the location but the accuracy is above threshold, it never fetches the location again.